htmresearch/frameworks/layers/simple_object_machine.py

```diff
@@ -253,12 +253,12 @@ def _addNoise(self, pattern, noiseLevel):
     if pattern is None:
       return None
 
-    newBits = []
+    newBits = set()
     for bit in pattern:
       if random.random() < noiseLevel:
-        newBits.append(random.randint(0, max(pattern)))
+        newBits.add(random.randint(0, self.externalInputSize))
```

Comment:
Oops, this is still not right. This size is different based on whether you're adding noise to the feature or the location.